### ISSUE TITLE
How to setup Skype Web SDK to accept login from database of Users from personal db

### ISSUE THREAD

**Issue by Onotseike** [OfficeDev/skype-web-sdk-samples#21]
Hi all, I have an issue. I am a newbie to the Skype Web SDK and O365 development. I am trying to integrate the Skype web sdk with my ASP.NET Core Web app. I have registered and confirmed ASPNETUsers in my Database, I want to setup the Skype Web SDK to enable skype once a user is signed into the app and he/she can talk with another registered user in the ASPNETUser database table. From the tutorials I have seen, it caters to only office 365 work accounts or special domain email addresses. 
Please any help will be greatly appreciated.

**Comment by sivijaya** [comment]
How do users log in to your web application? What credentials do they use?
Are these users also Skype for Business users (with valid Skype for Business licenses)?
Typically if the user is hosted on Skype for Business online then you can use Azure Active Directory to perform auth like documented. For onpremises users you could user username/password to perform authentication.

**Comment by Onotseike** [comment]
They login via ASP.NET Core Identity individual accounts,
These users are not Skype for business Users.
I am trying to create a Therapist Patient styled web app. the Therapists and patients are users with different roles under ASP.NET Identity. 

> Typically if the user is hosted on Skype for Business online then you can use Azure Active Directory to perform auth like documented. For onpremises users you could user username/password to perform authentication.

Could you explain how a user or group of users can be hosted in Skype for Business online. And could you explain what it entails for on-Premises users and how I can carry out authentication in the skype web sdk?

Many thanks for help.

**Comment by sivijaya** [comment]
See details on skype for business [here](https://www.skype.com/en/business/skype-for-business/). This requires purchasing office 365 subscriptions for users and enabling Skype for Business for each of those users. Typically an organization purchases Skype for Business licenses for all its users.
It sounds like you are trying to build a variation of a chat application? I'm not sure what ASP.NET Core Identity is. Do you create these user accounts?

**Comment by Onotseike** [comment]
> It sounds like you are trying to build a variation of a chat application? I'm not sure what ASP.NET Core Identity is. Do you create these user accounts?


Yes, I am trying to build an online therapy session app that has video, audio on messaging btw Patients and Therapists.
and Yes ASP.NET Core Identity is used to create accounts and add them to my Database. What I am confused about is how to authenticate the Skype web sdk when a patient logs in to begin chatting or calling .

**Comment by sivijaya** [comment]
Your users will have to sign in your application using their Skype for Business accounts. The process is similar to how an ASP.NET app authenticates with external providers (documented [here](https://docs.microsoft.com/en-us/aspnet/core/security/authentication/social/)) except that for Skype for Business you would follow steps like [this](https://msdn.microsoft.com/en-us/skype/websdk/docs/developwebsdkappsforsfbonline).
If you follow this approach you don't have any user credentials to store in your database - your web app redirects to Azure Active Directory's authentication page where your user enters their credentials. Then Azure AD will redirect back to your application with the user's token. The link that I shared above should have details on how to enable this flow for Skype for Business Online.

**Comment by Onotseike** [comment]
The thing is my users are regular people who don't own skype for Business accounts and I need them registered in my DB. Is it possible to use the On-Premises option to achieve this?

**Comment by sivijaya** [comment]
No, sorry, if you want your users to communicate with each other using Skype for Business then they need to have Skype for Business accounts. You may want to explore other free communication protocols for this purpose.

**Comment by Onotseike** [comment]
Aww, bummer. Could recommend any free communication protocols?
Many thanks for your help.

**Comment by satishakumar** [comment]
I too have similar scenario where web application uses MYSQL DB to store registered users. This web application is integrate with Skype for business online (Web SDK) to communicate each other via audio, video, chat and etc.
Functionality should be such a way that all the registered users(in MYSQL) should be able to access Skype Web SDK automatically without asking them to provide O 365 credentials. Because the users are already  registered for my application, asking them to provide O365 credentials to Skype services is not a good idea. 

So I want to build functionality such a way that when registered user(MYSQL) clicks say on '**Audio Call**', in background three things should happen automatically.  1.Creation of user in O 365 if not existed, 2. Sign in with O 365 credentials, 3. Start audio call.

Is this possible implement with Skype web SDK(online) ?

I am new to Skype for business online thing, Kindly suggest if any other ways available.
and also let me about available Web APIs to bulk create O 365 accounts using Global administrator of tenant.

**Comment by PrakashSivakumar** [comment]
I have a similar need to integrate Skype SDK in my ASP.Net Web application. Please help with the direction I'm supposed to take to have Audio, Video and Chat functionality just with authentication of the application login.
